pkg/runner/runner.go

```diff
@@ -0,0 +1,224 @@
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// EventTime - time to unmarshal nano time.
+type EventTime struct {
+	time.Time
+}
+
+// UnmarshalJSON - override unmarshal json.
+func (e *EventTime) UnmarshalJSON(b []byte) (err error) {
+	e.Time, err = time.Parse("2006-01-02T15:04:05.999999999", strings.Trim(string(b[:]), "\"\\"))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// MarshalJSON - override the marshal json.
+func (e EventTime) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf("\"%s\"", e.Time.Format("2006-01-02T15:04:05.99999999"))), nil
+}
+
+// JobEvent - event of an ansible run.
+type JobEvent struct {
+	UUID      string                 `json:"uuid"`
+	Counter   int                    `json:"counter"`
+	StdOut    string                 `json:"stdout"`
+	StartLine int                    `json:"start_line"`
+	EndLine   int                    `json:"EndLine"`
+	Event     string                 `json:"event"`
+	EventData map[string]interface{} `json:"event_data"`
+	PID       int                    `json:"pid"`
+	Created   EventTime              `json:"created"`
+}
+
+// StatusJobEvent - event of an ansible run.
+type StatusJobEvent struct {
+	UUID      string         `json:"uuid"`
+	Counter   int            `json:"counter"`
+	StdOut    string         `json:"stdout"`
+	StartLine int            `json:"start_line"`
+	EndLine   int            `json:"EndLine"`
+	Event     string         `json:"event"`
+	EventData StatsEventData `json:"event_data"`
+	PID       int            `json:"pid"`
+	Created   EventTime      `json:"created"`
+}
+
+// StatsEventData - data for a the status event.
+type StatsEventData struct {
+	Playbook     string         `json:"playbook"`
+	PlaybookUUID string         `json:"playbook_uuid"`
+	Changed      map[string]int `json:"changed"`
+	Ok           map[string]int `json:"ok"`
+	Failures     map[string]int `json:"failures"`
+	Skipped      map[string]int `json:"skipped"`
+}
+
+// Runner - a runnable that should take the parameters and name and namespace
+// and run the correct code.
+type Runner interface {
+	Run(map[string]interface{}, string, string, string) (*StatusJobEvent, error)
+}
+
+// Playbook - playbook type of runner.
+type Playbook struct {
+	Path string
+	GVK  schema.GroupVersionKind
+}
+
+// Run - This should allow the playbook runner to run.
+func (p *Playbook) Run(parameters map[string]interface{}, name, namespace, kc string) (*StatusJobEvent, error) {
+	parameters["meta"] = map[string]string{"namespace": namespace, "name": name}
+	runnerSandbox := fmt.Sprintf("/home/ansible-operator/runner/%s/%s/%s/%s/%s", p.GVK.Group, p.GVK.Version, p.GVK.Kind, namespace, name)
+	b, err := json.Marshal(parameters)
+	if err != nil {
+		return nil, err
+	}
+	//Write parameters to correct file on disk
+	err = createRunnerEnvironment(b, runnerSandbox, kc)
```

Comment:
What is kc? I think some of these short variable names make the code a little hard to follow.